lmdeploy/cli/serve.py

```diff
@@ -135,6 +135,9 @@ def add_parser_api_server():
 
         # chat template args
         ArgumentHelper.chat_template(parser)
+        ArgumentHelper.tool_call_parser(parser)
+        ArgumentHelper.enable_reasoning(parser)
+        ArgumentHelper.reasoning_parser(parser)
```

Comment:
then, we should deal cases and raise error or output messages in here when 
1. `reasoning_parser` is given, but `enable_reasoning` is not abled.
2. both `tool_call_parser` and `reasoning_parser` is given.

Reply:
`enable_reasoning` to me is a redundant option. It was added only because of vllm.

Reply:
> 2\. both `tool_call_parser` and `reasoning_parser` are given

Then only tool_call will be enabled.